model/example_domain_test.go

```diff
@@ -0,0 +1,618 @@
+package model_test
```

Comment:
There are no examples for `Domain`, as you can see from the first picture.

<img width="775" alt="image" src="https://user-images.githubusercontent.com/42215971/187498710-73a237c8-564a-40c7-8ce1-6977c7bd3fe5.png">

The idea is that these mirror the examples already present for `store.Domain`, so they feel the same. As the [Conventions](https://github.com/nextmv-io/plugins#conventions) state:

> To document examples for interface methods, name the test `func Example<Interface>_<method>()`. This will render the examples alongside the interface. Here is an [example from `store`](https://github.com/nextmv-io/sdk/blob/a0dbeac8f5d41e65e8fdb59cbcd679d533b77b41/store/example_domain_test.go#L10-L41):

```go
func ExampleDomain_add() {
	s1 := store.New()
	d := store.Multiple(s1, 1, 3, 5)
	s2 := s1.Apply(d.Add(2, 4))
	fmt.Println(d.Domain(s1))
	fmt.Println(d.Domain(s2))
	// Output:
	// {[{1 1} {3 3} {5 5}]}
	// {[{1 5}]}
}


func ExampleDomain_atLeast() {
	s1 := store.New()
	d := store.NewDomain(s1, model.NewRange(1, 10), model.NewRange(101, 110))
	s2 := s1.Apply(d.AtLeast(50))
	fmt.Println(d.Domain(s1))
	fmt.Println(d.Domain(s2))
	// Output:
	// {[{1 10} {101 110}]}
	// {[{101 110}]}
}


func ExampleDomain_atMost() {
	s1 := store.New()
	d := store.NewDomain(s1, model.NewRange(1, 10), model.NewRange(101, 110))
	s2 := s1.Apply(d.AtMost(50))
	fmt.Println(d.Domain(s1))
	fmt.Println(d.Domain(s2))
	// Output:
	// {[{1 10} {101 110}]}
	// {[{1 10}]}
}
```

You can see that the examples are rendered alongside the `Domain` interface:

<img width="727" alt="image" src="https://user-images.githubusercontent.com/42215971/187499995-e673f6be-b6b8-48ba-acfd-f62676118239.png">